src/sources/internal_metrics.rs

```diff
@@ -44,12 +33,14 @@ impl SourceConfig for InternalMetricsConfig {
         shutdown: ShutdownSignal,
         out: Pipeline,
     ) -> crate::Result<super::Source> {
-        Ok(Box::pin(run(
-            get_controller()?,
-            self.scrape_interval_secs,
-            out,
-            shutdown,
-        )))
+        let interval = time::Duration::from_secs(self.scrape_interval_secs);
+        if interval < time::Duration::from_millis(500) {
```

Comment:
```suggestion
        if interval < time::Duration::from_secs(1) {
```

Not a big deal, but I think this would be more consistent with the use of `u64` here.

Reply:
Shouldn't this be <= 0 to match the warning?

Reply:
Added https://github.com/timberio/vector/pull/5682/commits/874edc53c855f310460c888b2683a7ba92c99a78 instead

Reply:
:+1: that works too.